serializability-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializabilityCheckerCompilerPluginComponent.scala

```diff
@@ -18,52 +20,92 @@ class SerializabilityCheckerCompilerPluginComponent(
-                    s"""${next
-                      .toString()} is used as Akka message but does not extend a trait annotated with ${serializabilityTraitType.toLongString}.
-                      |Passing an object NOT extending ${serializabilityTraitType.nameAndArgsString} as a message may cause Akka to fall back to Java serialization during runtime.
-                      |Annotate this class or one of the traits/classes it extends with @${serializabilityTraitType.toLongString}.
-                      |
-                      |""".stripMargin)
+        val typesFromGenerics = if (pluginOptions.detectionFromGenerics) body.collect {
+          case x: TypeTree if genericsNames.contains(x.tpe.typeSymbol.fullName) =>
+            x.tpe.typeArgs.zip(genericsNamesWithTypes(x.tpe.typeSymbol.fullName))
+        }
+        else Nil
+
+        val typesFromGenericMethods = if (pluginOptions.detectionFromGenericMethods) body.collect {
+          case x: TypeApply if genericMethods.contains(x.symbol.fullName) =>
+            x.args.map(_.tpe).zip(genericMethodsWithTypes(x.symbol.fullName))
+        }
+        else Nil
+
+        val typesFromConcreteMethods = if (pluginOptions.detectionFromMethods) body.collect {
+          case x: Apply if concreteMethods.contains(x.symbol.fullName) =>
+            x.args.map(_.tpe).zip(concreteMethodsWithTypes(x.symbol.fullName))
+        }
+        else Nil
+
+        val foundTypes =
+          (typesFromGenerics ::: typesFromGenericMethods ::: typesFromConcreteMethods).flatten
+            .groupBy(_._1)
+            .map(_._2.head)
+
+        if (pluginOptions.verbose && foundTypes.nonEmpty) {
+          val fqcns = foundTypes.map(_._1.typeSymbol.fullName)
+          reporter.echo(body.pos, s"Found serializable types: ${fqcns.mkString(", ")}")
+        }
+
+        annotatedTraitsCache = foundTypes.foldRight(annotatedTraitsCache) { (next, annotatedTraits) =>
+          val (tpe, classType) = next
+          if (annotatedTraits.exists(tpe <:< _) || tpe.dealias.typeSymbol.fullName.startsWith("akka.")) {
+            annotatedTraits
+          } else {
+            findSuperclassAnnotatedWithSerializabilityTrait(tpe) match {
+              case Some(annotatedType) =>
+                if (annotatedTraits.contains(annotatedType)) {
                   annotatedTraits
-              }
+                } else {
+                  if (pluginOptions.verbose) {
+                    reporter.echo(
+                      s"${classOf[SerializabilityCheckerCompilerPlugin].getSimpleName}: Found new annotated trait: ${annotatedType.typeSymbol.fullName}")
+                  }
+                  annotatedType :: annotatedTraits
+                }
+              case None =>
+                reporter.error(
+                  tpe.typeSymbol.pos,
+                  s"""${tpe
+                    .toString()} is used as Akka ${classType.name} but does not extend a trait annotated with ${serializabilityTraitType.toLongString}.
```

Comment:
Hmm I'm getting the following error on the current `develop` commit of Hydra:

```
[error] Any is used as Akka message but does not extend a trait annotated with org.virtuslab.ash.SerializabilityTrait.
[error] Passing an object NOT extending SerializabilityTrait as a message may cause Akka to fall back to Java serialization during runtime.
[error] Annotate this class or one of the traits/classes it extends with @org.virtuslab.ash.SerializabilityTrait.
```

This is a potentially valid error... but the problem is that the message gives absolutely no clue as to where to look for the source of the error :/

Reply:
Very weird, looking into it

Reply:
It compiled without problem on my machine 🤔